src/Libraries/Microsoft.Extensions.AI/ChatCompletion/ChatClientStructuredOutputExtensions.cs

```diff
@@ -116,20 +115,37 @@ public static async Task<ChatCompletion<T>> CompleteAsync<T>(
         ChatOptions? options = null,
         bool? useNativeJsonSchema = null,
         CancellationToken cancellationToken = default)
-        where T : class
     {
         _ = Throw.IfNull(chatClient);
         _ = Throw.IfNull(chatMessages);
         _ = Throw.IfNull(serializerOptions);
 
         serializerOptions.MakeReadOnly();
 
-        var schemaNode = AIJsonUtilities.CreateJsonSchema(
+        var schemaElement = AIJsonUtilities.CreateJsonSchema(
             type: typeof(T),
             serializerOptions: serializerOptions,
             inferenceOptions: _inferenceOptions);
 
-        var schema = JsonSerializer.Serialize(schemaNode, AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(JsonElement)));
+        var isWrappedInObject = false;
+
+        if (!SchemaRepresentsObject(schemaElement))
+        {
+            // We wrap regardless of native structured output, since it also applies to Azure Inference
+            var schemaAsJsonObject = (JsonObject)JsonSerializer.Deserialize(
+                schemaElement,
+                AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(JsonObject)))!;
+            schemaElement = JsonSerializer.SerializeToElement(new JsonObject
+            {
+                { "$schema", "https://json-schema.org/draft/2020-12/schema" },
+                { "type", "object" },
+                { "properties", new JsonObject { { "data", schemaAsJsonObject } } },
```

Comment:
You can avoid creating an intermediate JsonObject representation for the schema:
```suggestion
                { "properties", new JsonObject { { "data", schemaElement } } },
```

Reply:
We can't do that exactly as it would not build:

```
Argument 2: cannot convert from 'System.Text.Json.JsonElement' to 'System.Text.Json.Nodes.JsonNode?'
```

I'll check your other suggestions in case that leads to a different solution.

Reply:
Pretty sure we had an implicit conversion for that, but seems we don't. `JsonValue.Create(schemaElement)` should work.

Reply:
Update: as you've pointed out, we only need to produce a `string`, so we can just do this using string interpolation instead of building a `JsonElement`.

Reply:
Shouldn't we still be honoring indentation configuration as present on the run-time JsonSerializerOptions?

Reply:
> Pretty sure we had an implicit conversion for that, but seems we don't. JsonValue.Create(schemaElement) should work.

Given this code:

```cs
schema = JsonSerializer.Serialize(new JsonObject
{
    { "$schema", "https://json-schema.org/draft/2020-12/schema" },
    { "type", "object" },
    { "properties", new JsonObject { { "data", JsonValue.Create(schemaElement) } } },
    { "additionalProperties", false },
}, AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(JsonElement)));
```

... it fails at runtime with:

```
    System.InvalidOperationException : The element cannot be an object or array.
```

Did I interpret your suggestion correctly or is there something else you mean?

Reply:
Updated as per other conversation